sdk/src/main/java/com/mapbox/maps/ResourceOptionsManager.kt

```diff
@@ -0,0 +1,48 @@
+package com.mapbox.maps
+
+import android.content.Context
+
+/**
+ * Convenience class that holds the default ResourceOptions.
+ * `ResourceOptionsManager` could be created per `MapView` instance if given view should use some specific resource options.
+ * Most common use-case when all `MapView`s should use same resource options could be handled by using `ResourceOptionsManager.getDefault(context: Context)` static object.
+ *
+ * @property context the context .
+ */
+data class ResourceOptionsManager(private var context: Context) {
+  var resourceOptions = ResourceOptions.Builder()
+    .accessToken(CredentialsManager.default.getAccessToken(context))
+    .cachePath("${context.filesDir.absolutePath}/$DATABASE_NAME")
+    .cacheSize(DEFAULT_CACHE_SIZE) // 50 mb
+    .build()
+    private set
+
+  /**
+   * Update specific resource options in a closure.
+   */
+  fun update(block: ResourceOptions.Builder.() -> Unit) {
+    resourceOptions = resourceOptions.toBuilder().apply(block).build()
```

Comment:
I think we need to expose to constants additionally default `cachePath` and default `accessToken`. Otherwise if user firstly changes let's say `cachePath` and for next map view he would like to use the default one - there is no way to get it I believe.

Reply:
Thanks to the suggestions from https://github.com/mapbox/mapbox-maps-android/issues/329, the defaultParams extension function is added to MapOptions and ResourceOptions' builder.